pkg/monitoring/metrics.go

```diff
@@ -40,6 +46,36 @@ func SetReadyGauge(isReady bool) {
 	}
 }
 
+func setKubemacpoolDeployedGauge(isDeployed bool) {
+	if isDeployed {
+		kmpDeployedGauge.Set(1)
+	} else {
+		kmpDeployedGauge.Set(0)
+	}
+}
+
+func TrackMonitoringComponents(conf *cnao.NetworkAddonsConfigSpec, monitoringAvailable bool) error {
+	if !monitoringAvailable {
+		return nil
+	}
+
+	err := trackKubemacpoolMonitoring(conf)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func trackKubemacpoolMonitoring(conf *cnao.NetworkAddonsConfigSpec) error {
+	isKubemacpoolDeployed := false
+	if conf.KubeMacPool != nil {
+		isKubemacpoolDeployed = true
+	}
+	setKubemacpoolDeployedGauge(isKubemacpoolDeployed)
```

Comment:
if you like the style, the whole function can be one line 
setKubemacpoolDeployedGauge(conf.KubeMacPool != nil)

Reply:
I prefer the variable with the long name `isKubemacpoolDeployed` to make the develper's life easier to understand..

Reply:
I am concerned about this hidden side effect. Could we either somehow lift this to an upper-level function or to rename this function to clearly show that it is changing the global? Maybe just use the setKubemacpoolDeployedGauge name directly? And if you could pass reference to the global to the function, it would get rid of the side effect completely

Reply:
I see, I'll move it upward, but not sure why I should pass the global. can you elaborate?

Reply:
If you pass link to the global, you can unit test it. It would also become clear what is being "set". With it being taken from the global context, we cannot easily unit test it. It is unclear what this function works with unless you peek into it.

Reply:
Sounds good